test_runpy fails on non-UTF-8 locale

```
$ LC_ALL=en_US.iso885915 ./python -m test -vuall test_runpy -m test_main_recursion_error
== CPython 3.11.0a7+ (heads/re-group-name:5026649f98, Apr 24 2022, 22:45:28) [GCC 9.4.0]
== Linux-5.4.0-109-generic-x86_64-with-glibc2.31 little-endian
== cwd: /home/serhiy/py/cpython/build/test_python_197609�
== CPU count: 8
== encodings: locale=ISO-8859-15, FS=iso8859-15
0:00:00 load avg: 0.34 Run tests sequentially
0:00:00 load avg: 0.34 [1/1] test_runpy
test_main_recursion_error (test.test_runpy.RunPathTestCase.test_main_recursion_error) ... FAIL

======================================================================
FAIL: test_main_recursion_error (test.test_runpy.RunPathTestCase.test_main_recursion_error)
----------------------------------------------------------------------
RecursionError: maximum recursion depth exceeded

The above exception was the direct cause of the following exception:

RecursionError

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/home/serhiy/py/cpython/Lib/test/support/__init__.py", line 1053, in wrapper
    return func(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^
  File "/home/serhiy/py/cpython/Lib/contextlib.py", line 155, in __exit__
    self.gen.throw(typ, value, traceback)
    ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/serhiy/py/cpython/Lib/test/support/os_helper.py", line 396, in temp_dir
    yield path
    ^^^^^^^^^^
  File "/home/serhiy/py/cpython/Lib/contextlib.py", line 155, in __exit__
    self.gen.throw(typ, value, traceback)
    ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/serhiy/py/cpython/Lib/test/support/os_helper.py", line 396, in temp_dir
    yield path
    ^^^^^^^^^^
  File "/home/serhiy/py/cpython/Lib/test/test_runpy.py", line 744, in test_main_recursion_error
    self.assertRaisesRegex(RecursionError, msg, run_path, zip_name)
    ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
AssertionError: "recursion depth exceeded" does not match ""

----------------------------------------------------------------------
Ran 1 test in 1.305s

FAILED (failures=1)
test test_runpy failed
test_runpy failed (1 failure)

== Tests result: FAILURE ==

1 test failed:
    test_runpy

Total duration: 1.4 sec
Tests result: FAILURE
```

--- a/Lib/test/test_runpy.py
+++ b/Lib/test/test_runpy.py
@@ -740,8 +740,7 @@ class RunPathTestCase(unittest.TestCase, CodeExecutionMixin):
                       "runpy.run_path(%r)\n") % dummy_dir
             script_name = self._make_test_script(script_dir, mod_name, source)
             zip_name, fname = make_zip_script(script_dir, 'test_zip', script_name)
-            msg = "recursion depth exceeded"
-            self.assertRaisesRegex(RecursionError, msg, run_path, zip_name)
+            self.assertRaises(RecursionError, run_path, zip_name)
 
     def test_encoding(self):
         with temp_dir() as script_dir:

